base/math.jl

```diff
@@ -128,15 +128,17 @@ end
 @inline function exp2(x::Base.BitInteger)
     if x > 1023
         Inf64
-    elseif x < -1074
-        Float64(0.0)
     elseif x <= -1023
-        # Result will be a subnormal number
-        reinterpret(Float64, Int64(1) << (x + 1074))
+        if x < -1074
+            # Result will be a subnormal number
+            reinterpret(Float64, (1 % Int64) << (x + 1074))
```

Comment:
You could convert the shift count to `unsigned` via `%`; otherwise Julia will have to check the sign.